src/mk.sh

```diff
@@ -482,6 +483,8 @@ function kernel_deploy
     exit 125 # ECANCELED
   fi
 
+  signal_manager 'cleanup; exit 0'
```

Comment:
Hi @joaoseckler ,
In our previous conversation, I pointed out that this could be a problem. Maybe you can create a simple signal manager in the mk file that will detect which option the user used (you just need to take a look at `options_values`) and run a specific action.

Btw, remove the `Draft` from your PR. Your PR is excellent.